utils/src/registerCommand.ts

```diff
@@ -19,17 +24,18 @@ export function registerCommand(commandId: string, callback: (context: types.IAc
             }
             lastClickTime = Date.now();
         }
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
         return await callWithTelemetryAndErrorHandling(
             telemetryId || commandId,
-            (context: types.IActionContext) => {
+            async (context: types.IActionContext) => {
                 if (args.length > 0) {
                     const firstArg: unknown = args[0];
 
-                    if (firstArg instanceof AzExtTreeItem) {
-                        context.telemetry.properties.contextValue = firstArg.contextValue;
-                    } else if (firstArg instanceof Uri) {
+                    if (firstArg instanceof Uri) {
                         context.telemetry.properties.contextValue = 'Uri';
+                    } else if (isTreeElementBase(firstArg)) {
+                        context.telemetry.properties.contextValue = (await firstArg.getTreeItem()).contextValue;
```

Comment:
If we still want to base the command source from if the item has a contextValue, then we need to do it like this. 

I'm not opposed to changing this, but it'd mean we'd have to change a bunch of queries in the dashboard.

Reply:
Can we flip the order around, i.e. do `else if (isAzExtTreeItem(firstArg))` _before_ doing this check involving `getTreeItem()`?

Reply:
In fact, could we cheat a little and just see if the element has a `contextValue` property that is a `string`, and just use that if present, otherwise do the `getTreeItem()`-based test? Theoretically that could be less accurate but for the sake of telemetry I think it's good enough.